sdk/servicebus/azure-servicebus/azure/servicebus/_common/_configuration.py

```diff
@@ -6,10 +6,24 @@
 
 from uamqp.constants import TransportType
 
+from .constants import MAX_USER_AGENT_LENGTH
+
 
 class Configuration(object):  # pylint:disable=too-many-instance-attributes
     def __init__(self, **kwargs):
         self.user_agent = kwargs.get("user_agent")  # type: Optional[str]
+
+        if self.user_agent:
+            if len(self.user_agent) > MAX_USER_AGENT_LENGTH:
+                raise ValueError(
+                    "The user-agent string cannot be more than {} in length."
+                    "The length of the provided string is: {}".format(
+                        MAX_USER_AGENT_LENGTH, len(self.user_agent)
+                    )
+                )
+            if ' ' in self.user_agent:
```

Comment:
I feel a little bit leery about doing "some validation" here but not all, (since I imagine stuff like \n and \t is also invalid) 

I assume this is because there's no service-side error that makes this at all clear?  if so I can live with that.  May be useful in the future to check for a regex using \s for "all whitespace" but consider this comment "aspirational" and we can ignore this for now unless it strongly moved you.

Reply:
pending user-agent arch board meeting, it would be best if we could follow how azure-core do validation.

Reply:
@KieranBrantnerMagee , after OOB discussion with @xiangyan99 , we're not going to validate user_agent (following azure-core) until further decision has been made.

Reply:
Great, this would seem to align with our broader principles on validation.  Thanks for checking up on that.